buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -655,6 +655,7 @@ class BuildPlugin implements Plugin<Project> {
                     project.mkdir(testOutputDir)
                     project.mkdir(heapdumpDir)
                     project.mkdir(test.workingDir)
+                    project.mkdir(test.workingDir.toPath().resolve('temp'))
```

Comment:
@mark-vieira Am I missing something why this isn't necessary in other tests? Without this, in the `:distribution:tools:launchers:test` tests, the directory wouldn't exist on a clean run and the tests would fail?

Reply:
Since this project seems to be a one-off, and the lucene test framework does this for all our other projects, I'm inclined to just do this in the build script for the launch project instead of adding this to `BuildPlugin`. Or put the smarts for this in the launcher tests themselves.

Reply:
I'm not sure. It's a bit weird that we set this as the temporary directory for tests (via `java.io.tmpdir`) but we rely on a test case base class (not even our test runner) to ensure the directory exists:

https://github.com/elastic/elasticsearch/blob/dbf08eb39550d3da2cbf3949c2492e09a6c0e8e6/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy#L691

It seems that we would want this directory to always exists, relying on that.

Reply:
That's fair. Given that, let's go ahead and move the creation of that directory to where we set `java.io.tmpdir` so that logic is colocated. I'd also prefer we use an absolute path here rather than relying on implicit behavior on this being relative to the test working directory.

Reply:
Note that that would be moving the creation of the temporary directory into test task configuration, it seems we wouldn't want that? Right now I've put it in a `doFirst` so it's only created upon execution.

There might be a reason for the relative path, I'm still looking into that.

Reply:
> Note that that would be moving the creation of the temporary directory into test task configuration, it seems we wouldn't want that? Right now I've put it in a doFirst so it's only created upon execution.

You are 100% right. Disregard 😄

Reply:
It *looks* like the relative nature of the path is simply leftover legacy from the Maven days when we didn't know the working directory.

So I think we can go absolute, but I would spin this out into a separate change now, so it's explicit. I dislike when changes like this are worked into another PR as a side story when they should be a main dish all on their own for others to see.

Reply:
I opened #52227 and #52228.